charters/UploadService/charter.md

```diff
@@ -0,0 +1,80 @@
+# Upload Service
+
+## Description
+The Human Cell Atlas (HCA) Data Coordination Platform (DCP) Upload Service provides a file staging and validation facility. Upload Areas are created and deleted using a [REST](https://en.wikipedia.org/wiki/Representational_state_transfer) Application Programming Interface (API), which is secured so only the DCP Ingestion Service may use it. It stages files into cloud storage and computes checksums for the files. The validation service can run user-defined validation jobs against files.
+
+## Objectives
+1. Provide temporary storage for staging of files by submitters during ingestion into DCP Ingestion Service and subsequent storage in the Data Storage Service (DSS)
+1. Provide tools a command line interface (CLI) and API to upload files to staging areas
+1. Provide a facility to validate files in staging area that accepts user-defined validators that meet upload service convention
+
+## In-scope
+* Staging Areas - provide file staging areas to be used to marshal data and metadata files
+    * Efficient file transfer to the DCP-DSS, the data’s final location
+    * Efficient file transfer from submitters in remote geographic locations
+    * Efficient file transfer from the DCP Secondary Analysis Service (which deposits results in GCS)
+* HTTP REST API
+    * Functionality for other DCP modules, namely Ingest
+        * Request creation and deletion of staging areas
+        * Schedule and retrieve validation jobs for files
+        * Store metadata content into staging area
+    * Functionality for Submitters
+        * Obtain credentials to write to upload areas
+        * Examine the contents of upload areas
+* Checksumming
+    * Compute `SHA-1`, `SHA-256`, `CRC-32C` and `S3 ETAG` checksums for all staged files and make these checksums available to the DSS
```

Comment:
It may be enough to generically indicate checksums in the charter and not specifics, and then implement the specific ones as adopted by the system.